e2e/testsuite/tx.go

```diff
@@ -97,9 +97,9 @@ func containsMessage(s string, messages []string) bool {
 
 // AssertTxFailure verifies that an sdk.TxResponse has failed.
 func (s *E2ETestSuite) AssertTxFailure(resp sdk.TxResponse, expectedError *errorsmod.Error) {
+	// In older versions, the codespace and abci codes were different. So in compatibility tests
+	// we can not make assertions on them.
 	errorMsg := fmt.Sprintf("%+v", resp)
-	s.Require().Equal(expectedError.ABCICode(), resp.Code, errorMsg)
-	s.Require().Equal(expectedError.Codespace(), resp.Codespace, errorMsg)
```

Comment:
could the arg now be built-in `error`? happy to keep it this way

Reply:
It'd be nice to be able to keep this functionality if possible. Could we skip only for compatibility? Maybe we could comment them out for now?

Reply:
Opened #3652